Chessington.GameEngine/Pieces/Knight.cs

```diff
@@ -10,7 +10,27 @@ public Knight(Player player)
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            List<Square> availableMoves = new List<Square>();
+            
+            int[] row = new int[] { 2, 1, -1, -2, -2, -1, 1, 2};
+            int[] col = new int[] { 1, 2, 2, 1, -1, -2, -2, -1};
+
+            for (int i = 0; i < 8; i++)
+            {
+                if (board.FindPiece(this).Row + row[i] >= 0
+                    && board.FindPiece(this).Row + row[i] < 8
+                    && board.FindPiece(this).Col + col[i] >= 0
+                    && board.FindPiece(this).Col + col[i] < 8 
+                    && (board.GetPiece(new Square(board.FindPiece(this).Row + row[i],
```

Comment:
You're repeating "board.FindPiece(this).Row + row[i]" quite a few times. I'd suggest using a variable instead